autonat/autonat-v2.md

```diff
@@ -127,37 +151,40 @@ message Message {
     DIAL_ATTEMPT  = 2;
   }
 
-  enum ResponseStatus {
-    OK                        = 0;
-    E_DIAL_ERROR              = 100;
-    E_DIAL_REFUSED            = 101;
-    E_TRANSPORT_NOT_SUPPORTED = 102;
-    E_BAD_REQUEST             = 200;
-    E_INTERNAL_ERROR          = 300;
-  }
+  MessageType type          = 1;
+  DialRequest dialRequest   = 2;
+  DialResponse dialResponse = 3;
+  DialAttempt dialAttempt   = 4;
+}
 
-  message DialRequest {
-    bytes addr = 1;
-    fixed64 nonce = 2;
-  }
+message AddressDialRequest {
+  bytes addr = 1;
+  fixed64 nonce = 2;
+}
 
-  message DialResponse {
-    ResponseStatus status = 1;
-    string statusText = 2;
-  }
+message DialRequest {
+  repeated AddressDialRequest addressDialRequests = 1;
+}
 
-  message DialAttempt {
+message DialAttempt {
     fixed64 nonce = 1;
-  }
+}
 
-  MessageType type = 1;
-  DialRequest dialRequest = 2;
-  DialResponse dialResponse = 3;
-  DialAttempt dialAttempt = 4;
+message DialResponse {
+  enum ResponseStatus {
+      OK                        = 0;
+      E_DIAL_ERROR              = 100;
+      E_DIAL_REFUSED            = 101;
+      E_TRANSPORT_NOT_SUPPORTED = 102;
+      E_BAD_REQUEST             = 200;
+      E_INTERNAL_ERROR          = 300;
+    }
+
+    ResponseStatus status = 1;
+    string statusText = 2;
+    bytes addr = 3;
```

Comment:
Is there any advantage to sending the entire address? This could be the list index of the address in the `DialRequest`.

Reply:
changing this to index, I think that would also be easier for implementations